build.assets/kubectl-version/main.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"fmt"
+	"runtime/debug"
+	"strings"
+
+	_ "k8s.io/kubectl"
+)
+
+func main() {
+	if version, ok := getKubectlModVersion(); ok {
```

Comment:
Can you write this to exit with a non-zero error code if it fails? Otherwise there is no clear indication that this failed during a build.